### PR TITLE
Move DAV app to migrations

### PR DESCRIPTION
**Pull request by rullzer**
Need to touch the db schema for https://github.com/nextcloud/server/issues/6170
So better first move to migrations.

### CODE REVIEW

**Comment by nickvergessen** [comment]
Version on the migration should be `1004` and then the version in info.xml should be increased

**Comment by rullzer** [comment]
@nickvergessen done

**Review by MorrisJobke — APPROVED** [review]
Tested update and install. Both resulted in a working DB structure. 👍

File: apps/dav/lib/Migration/Version1004001Date20170825134824.php [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+			]);
+			$table->addColumn('uri', 'string', [
+				'notnull' => false,
+				'length' => 255,
+			]);
+			$table->addColumn('description', 'string', [
+				'notnull' => false,
+				'length' => 255,
+			]);
+			$table->addColumn('synctoken', 'integer', [
```
Comment (nickvergessen) [inline apps/dav/lib/Migration/Version1004001Date20170825134824.php]: this changed from 10 to 11 chars Oo
Also all int columns dropped `UNSIGNED`

**Comment by nickvergessen** [comment]
```diff
--- /home/nickv/Schreibtisch/nextcloud13.sql
+++ /home/nickv/Schreibtisch/nextcloud13b.sql
@@ -38,9 +38,9 @@
 --
 
 CREATE TABLE `oc_addressbookchanges` (
-  `id` bigint(20) UNSIGNED NOT NULL,
-  `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `synctoken` int(10) UNSIGNED NOT NULL DEFAULT '1',
+  `id` bigint(20) NOT NULL,
+  `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
+  `synctoken` int(11) NOT NULL DEFAULT '1',
   `addressbookid` int(11) NOT NULL,
   `operation` smallint(6) NOT NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
@@ -52,12 +52,12 @@
 --
 
 CREATE TABLE `oc_addressbooks` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `principaluri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `displayname` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `description` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `synctoken` int(10) UNSIGNED NOT NULL DEFAULT '1'
+  `synctoken` int(11) NOT NULL DEFAULT '1'
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
 
 -- --------------------------------------------------------
@@ -67,9 +67,9 @@
 --
 
 CREATE TABLE `oc_calendarchanges` (
-  `id` bigint(20) UNSIGNED NOT NULL,
-  `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `synctoken` int(10) UNSIGNED NOT NULL DEFAULT '1',
+  `id` bigint(20) NOT NULL,
+  `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
+  `synctoken` int(11) NOT NULL DEFAULT '1',
   `calendarid` int(11) NOT NULL,
   `operation` smallint(6) NOT NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
@@ -81,18 +81,18 @@
 --
 
 CREATE TABLE `oc_calendarobjects` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `calendardata` longblob,
   `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `calendarid` int(10) UNSIGNED NOT NULL,
-  `lastmodified` int(10) UNSIGNED DEFAULT NULL,
+  `calendarid` int(11) NOT NULL,
+  `lastmodified` int(11) DEFAULT NULL,
   `etag` varchar(32) COLLATE utf8mb4_bin DEFAULT NULL,
-  `size` bigint(20) UNSIGNED NOT NULL,
+  `size` bigint(20) NOT NULL,
   `componenttype` varchar(8) COLLATE utf8mb4_bin DEFAULT NULL,
-  `firstoccurence` bigint(20) UNSIGNED DEFAULT NULL,
-  `lastoccurence` bigint(20) UNSIGNED DEFAULT NULL,
+  `firstoccurence` bigint(20) DEFAULT NULL,
+  `lastoccurence` bigint(20) DEFAULT NULL,
   `uid` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `classification` int(11) DEFAULT '0' COMMENT '0 - public, 1 - private, 2 - confidential'
+  `classification` int(11) DEFAULT '0'
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
 
 -- --------------------------------------------------------
@@ -102,9 +102,9 @@
 --
 
 CREATE TABLE `oc_calendarobjects_props` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `calendarid` bigint(20) NOT NULL DEFAULT '0',
-  `objectid` bigint(20) UNSIGNED NOT NULL DEFAULT '0',
+  `objectid` bigint(20) NOT NULL DEFAULT '0',
   `name` varchar(64) COLLATE utf8mb4_bin DEFAULT NULL,
   `parameter` varchar(64) COLLATE utf8mb4_bin DEFAULT NULL,
   `value` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL
@@ -117,13 +117,13 @@
 --
 
 CREATE TABLE `oc_calendars` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `principaluri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `displayname` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `synctoken` int(10) UNSIGNED NOT NULL DEFAULT '1',
+  `synctoken` int(11) NOT NULL DEFAULT '1',
   `description` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `calendarorder` int(10) UNSIGNED NOT NULL DEFAULT '0',
+  `calendarorder` int(11) NOT NULL DEFAULT '0',
   `calendarcolor` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `timezone` longtext COLLATE utf8mb4_bin,
   `components` varchar(64) COLLATE utf8mb4_bin DEFAULT NULL,
@@ -137,18 +137,18 @@
 --
 
 CREATE TABLE `oc_calendarsubscriptions` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `principaluri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `source` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `displayname` varchar(100) COLLATE utf8mb4_bin DEFAULT NULL,
   `refreshrate` varchar(10) COLLATE utf8mb4_bin DEFAULT NULL,
-  `calendarorder` int(10) UNSIGNED NOT NULL DEFAULT '0',
+  `calendarorder` int(11) NOT NULL DEFAULT '0',
   `calendarcolor` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `striptodos` smallint(6) DEFAULT NULL,
   `stripalarms` smallint(6) DEFAULT NULL,
   `stripattachments` smallint(6) DEFAULT NULL,
-  `lastmodified` int(10) UNSIGNED DEFAULT NULL
+  `lastmodified` int(11) DEFAULT NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
 
 -- --------------------------------------------------------
@@ -158,13 +158,13 @@
 --
 
 CREATE TABLE `oc_cards` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `addressbookid` int(11) NOT NULL DEFAULT '0',
   `carddata` longblob,
   `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `lastmodified` bigint(20) UNSIGNED DEFAULT NULL,
+  `lastmodified` bigint(20) DEFAULT NULL,
   `etag` varchar(32) COLLATE utf8mb4_bin DEFAULT NULL,
-  `size` bigint(20) UNSIGNED NOT NULL
+  `size` bigint(20) NOT NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
 
 -- --------------------------------------------------------
@@ -174,9 +174,9 @@
 --
 
 CREATE TABLE `oc_cards_properties` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `addressbookid` bigint(20) NOT NULL DEFAULT '0',
-  `cardid` bigint(20) UNSIGNED NOT NULL DEFAULT '0',
+  `cardid` bigint(20) NOT NULL DEFAULT '0',
   `name` varchar(64) COLLATE utf8mb4_bin DEFAULT NULL,
   `value` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `preferred` int(11) NOT NULL DEFAULT '1'
@@ -189,11 +189,11 @@
 --
 
 CREATE TABLE `oc_dav_shares` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `principaluri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `type` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `access` smallint(6) DEFAULT NULL,
-  `resourceid` int(10) UNSIGNED NOT NULL,
+  `resourceid` int(11) NOT NULL,
   `publicuri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
 
@@ -204,13 +204,13 @@
 --
 
 CREATE TABLE `oc_schedulingobjects` (
-  `id` bigint(20) UNSIGNED NOT NULL,
+  `id` bigint(20) NOT NULL,
   `principaluri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
   `calendardata` longblob,
   `uri` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
-  `lastmodified` int(10) UNSIGNED DEFAULT NULL,
+  `lastmodified` int(11) DEFAULT NULL,
   `etag` varchar(32) COLLATE utf8mb4_bin DEFAULT NULL,
-  `size` bigint(20) UNSIGNED NOT NULL
+  `size` bigint(20) NOT NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED;
 
 --
```

**Comment by MorrisJobke** [comment]
@nickvergessen Mind to have another look?

**Review by nickvergessen — APPROVED** [review]

**Comment by rullzer** [comment]
I just rebased to make sure CI is green for this

**Pull request merged**